core/validatorapi/router.go

```diff
@@ -453,13 +460,42 @@ func attestationData(p eth2client.AttestationDataProvider) handlerFunc {
 // submitAttestations returns a handler function for the attestation submitter endpoint.
 func submitAttestations(p eth2client.AttestationsSubmitter) handlerFunc {
 	return func(ctx context.Context, _ map[string]string, _ url.Values, typ contentType, body []byte) (any, http.Header, error) {
-		var atts []*eth2spec.VersionedAttestation
-		err := unmarshal(typ, body, &atts)
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "unmarshal attestations")
+		atts := []*eth2spec.VersionedAttestation{}
+
+		denebAtts := new([]eth2p0.Attestation)
+		err := unmarshal(typ, body, denebAtts)
+		if err == nil {
+			for _, att := range *denebAtts {
+				// TODO: Data version is not Deneb, it might be anything between Phase0 and Deneb
```

Comment:
Flagging this TODO, maybe you forgot to resolve it.

Reply:
Unfortunately, I haven't found a nice way yet... It's in my list for things to fix.

With other versioned structures (like blocks), we have a clunky but functional way of resolving the version - we simply try to parse all versions and whichever succeeds, we deem it is that one. However, for attestations, anything between phase0 and Deneb has the exact same structure, so this method won't really work.